ssl/ssl_lib.c

```diff
@@ -294,9 +295,12 @@ static int dane_tlsa_add(SSL_DANE *dane,
         }
     }
 
-    if (md != NULL && dlen != (size_t)EVP_MD_get_size(md)) {
-        ERR_raise(ERR_LIB_SSL, SSL_R_DANE_TLSA_BAD_DIGEST_LENGTH);
-        return 0;
+    if (md != NULL) {
+        mdsize = EVP_MD_get_size(md);
+        if (mdsize < 0 || dlen != (size_t)mdsize) {
```

Comment:
`<= 0` as 0 md size is an error.

Reply:
Thanks. I have added a new commit to alter the check.